spec/migrations/20191023180330_migrate_ems_ref_obj_to_ems_ref_type_spec.rb

```diff
@@ -0,0 +1,115 @@
+require_migration
+
+describe MigrateEmsRefObjToEmsRefType do
+  let(:cluster_stub)       { migration_stub(:EmsCluster) }
+  let(:folder_stub)        { migration_stub(:EmsFolder) }
+  let(:host_stub)          { migration_stub(:Host) }
+  let(:resource_pool_stub) { migration_stub(:ResourcePool) }
+  let(:storage_stub)       { migration_stub(:Storage) }
+  let(:vm_stub)            { migration_stub(:VmOrTemplate) }
+
+  migration_context :up do
+    it "migrates ems_ref_obj to ems_ref_type" do
+      vm = vm_stub.create!(
+        :type        => "ManageIQ::Providers::Vmware::InfraManager::Vm",
+        :ems_ref     => "vm-123",
+        :ems_ref_obj => <<~EMS_REF_OBJ
+          --- !ruby/string:VimString
+          str: vm-1234
+          xsiType: :ManagedObjectReference
+          vimType: :VirtualMachine
+        EMS_REF_OBJ
+      )
+      host = host_stub.create!(
+        :type        => "ManageIQ::Providers::Vmware::InfraManager::HostEsx",
+        :ems_ref     => "host-123",
+        :ems_ref_obj => <<~EMS_REF_OBJ
+          --- !ruby/string:VimString
+          str: host-1234
+          xsiType: :ManagedObjectReference
+          vimType: :HostSystem
+        EMS_REF_OBJ
+      )
+      cluster = cluster_stub.create!(
+        :ems_ref     => "domain-c1234",
+        :ems_ref_obj => <<~EMS_REF_OBJ
+          --- !ruby/string:VimString
+          str: domain-c1234
+          xsiType: :ManagedObjectReference
+          vimType: :ClusterComputeResource
+        EMS_REF_OBJ
+      )
+      respool = resource_pool_stub.create!(
+        :ems_ref     => "resgroup-1234",
+        :ems_ref_obj => <<~EMS_REF_OBJ
+          --- !ruby/string:VimString
+          str: resgroup-1234
+          xsiType: :ManagedObjectReference
+          vimType: :ResourcePool
+        EMS_REF_OBJ
+      )
+      storage = storage_stub.create!(
+        :ems_ref     => "datastore-1234",
+        :ems_ref_obj => <<~EMS_REF_OBJ
+          --- !ruby/string:VimString
+          str: datastore-1234
+          xsiType: :ManagedObjectReference
+          vimType: :Datastore
+        EMS_REF_OBJ
+      )
+
+      migrate
+
+      expect(vm.reload.ems_ref_type).to      eq("VirtualMachine")
+      expect(host.reload.ems_ref_type).to    eq("HostSystem")
+      expect(cluster.reload.ems_ref_type).to eq("ClusterComputeResource")
+      expect(respool.reload.ems_ref_type).to eq("ResourcePool")
+      expect(storage.reload.ems_ref_type).to eq("Datastore")
+    end
+
+    it "ignores ems_ref_obj if it isn't a VimString" do
```

Comment:
Is it possible there are entires will nil values in ems_ref_obj?  Should there be a test for nil values?

Reply:
It would most likely be `"--- \n"` but I'll add a test for `nil` as well just in case

Reply:
[here](https://github.com/ManageIQ/manageiq-schema/pull/426/files#diff-e15ea0fdfded664940afbedd79eb324cR82)